test/e2e/generator/generate.go

```diff
@@ -84,7 +84,7 @@ const (
 // generateTestnet generates a single testnet with the given options.
 func generateTestnet(r *rand.Rand, opt map[string]interface{}) (e2e.Manifest, error) {
 	manifest := e2e.Manifest{
-		IPv6:             opt["ipv6"].(bool),
+		IPv6:             ipv6.Choose(r).(bool),
```

Comment:
Does this mean we randomly choose one or the other?

Reply:
yeah, we past in a `rand.Rand` (which is PRNG, etc.) with a const seed so that the testnets have consistent identities relative to any specific version of the generator, but yeah, you'd expect that this means that half are true and half are false.